### PR TITLE
Project 1 Part 1 - Anna Leong, basics-9-11

### PR DESCRIPTION
**Pull request by annaleongmy**
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment?
1.5h total

Please fill in one error and/or error message you received while working on this assignment.
When I first tried to do the input validation, I did if(input == !'scissors') instead of if(input != 'scissors')

What part of the assignment did you spend the most time on?
Trying to find a way to add emojis into my output elegantly.
It took awhile for me to realise I could just copy and paste the icon in because Google was going on about unicodes and got me confused. After that, I was trying to emulate the way the Hawker Dish Omakase solution had worked in the dish names/descriptions so the right emoji would pop up in the output depending on the gameplay.

Comfort Level (1-5): 4

Completeness Level (1-5): 5

What did you think of this deliverable?
The base level was really easy, so I was motivated to try the more comfortable level

Is there anything in this code that you feel pleased about?
I always feel pleased when my code works. The novelty has yet to wear off!

### CODE REVIEW

**Comment by annaleongmy** [comment]
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment?
Around 5h total, on and off

Please fill in one error and/or error message you received while working on this assignment.
At some point, the computer was not playing the game as my game logic function was running before the random generator function. At another point, the game seemed to be playing in the background even though the input validation message was showing, or I was trying to change the game mode. I realised the importance of the order of variables within a function, and also how to use return to "short-circuit" the rest of the function from running.

What part of the assignment did you spend the most time on?
After adding on so many different game modes, I realised there was some double counting in the score tracker. I think both the default game mode and alternative game modes were running at the same time. I added else conditions to make it clearer to the computer that I only wanted 1 game mode to run at any one time.

Comfort Level (1-5): 3

Completeness Level (1-5): 4

What did you think of this deliverable?
It was not easy but I learnt a lot in the process! One point of feedback is that there seems to be an auto code checker from part 1 of the project submission that sends me notifications each time I do a git push in part 2. It is a bit annoying and demoralising, because I'm not sure what all the feedback means and whether it is relevant at this stage. For example it tells me to consider shortening some of my code, when I am following the formatting being taught in the course so far. It would be better if we can run through that feedback together in class with proper context.

Is there anything in this code that you feel pleased about?
I managed to resolve the problems encountered (eventually)! And I was able to refactor some of the functions to make the main function look neater.

File: script.js [hunk 1]
```diff
@@ -1,4 +1,238 @@
...
+    var currentGameLogic = defaultGameLogic(userPlays, computerPlays);
+  }else{
+      if(gameMode == `reverse`){
+      currentGameLogic = reverseGameLogic(userPlays, computerPlays);
+      }else{
+        if(gameMode == `korean`){
+          currentGameLogic = koreanGameLogic(userPlays, computerPlays);
+          }else{
+            if(gameMode == `computer`){
+              currentGameLogic = defaultGameLogic(generateRandomSPS(), computerPlays);
```
Comment (michelle-mok) [inline script.js:103]: consider using multiple if/return here to simplify your code

File: script.js [hunk 2]
```diff
@@ -1,4 +1,238 @@
...
+                    return `Computer wins the game!`
+                  }else{
+                      numTimesDraw += 1;
+                      return `No attacker yet. Keep playing`;
+                  };
+            };
+          };
+      };
+  };
+};
```
Comment (michelle-mok) [inline script.js:201]: good use of nested if/else conditions!

File: script.js [hunk 3]
```diff
@@ -1,4 +1,238 @@
...
+
+//generate random number from 1 to 3
```
Comment (michelle-mok) [inline script.js:203]: these 3 functions should go at the top of your code because functions should be defined before they are called

**Review by michelle-mok — COMMENTED** [review]
excellent use of comments, variables are well named, code is well organised and easy to read. 
excellent use of functions to avoid repetition too!
well done! keep up the awesome work!

**Pull request closed without merging**